src/braket/circuits/result_type.py

```diff
@@ -209,32 +209,32 @@ def __init__(
         super().__init__(ascii_symbols)
         self._observable = observable
         self._target = QubitSet(target)
-        if not self._target:
-            if self._observable.qubit_count != 1:
-                raise ValueError(
-                    f"Observable {self._observable} must only operate on 1 qubit for target=None"
-                )
-        elif isinstance(observable, Sum):  # nested target
-            if len(target) != len(observable.summands):
-                raise ValueError(
-                    "Sum observable's target shape must be a nested list where each term's "
-                    "target length is equal to the observable term's qubits count."
-                )
-            self._target = [QubitSet(term_target) for term_target in target]
-            for term_target, obs in zip(self._target, observable.summands):
-                if obs.qubit_count != len(term_target):
+        if self._target:
+            if isinstance(observable, Sum):  # nested target
+                if len(target) != len(observable.summands):
                     raise ValueError(
                         "Sum observable's target shape must be a nested list where each term's "
                         "target length is equal to the observable term's qubits count."
                     )
-        elif self._observable.qubit_count != len(self._target):
-            raise ValueError(
-                f"Observable's qubit count {self._observable.qubit_count} and "
-                f"the size of the target qubit set {self._target} must be equal"
-            )
-        elif self._observable.qubit_count != len(self.ascii_symbols):
```

Comment:
Lines 217-237 are identical to lines 213-233 in the change; all I've done is indent one level